### DISCUSSION TITLE
Request for help in getting WebLinksAddon to work (how to activate/enable MouseEvents onClick?)

### DISCUSSION THREAD

**Discussion by isubasin** [sourcelair/xterm.js#4425] [General]
Hello, 

We are trying to get hyperlinks to work on an existing web terminal with references to help in the forums however, though the links are displayed, they are not selectable. Since our team is new xterm reaching out to anyone who can help us figure out what we are missing.

Observation: Mouse click just highlights the entire section but does not trigger any events. (Right-click options are working , normal left mouse click event is not triggering any action other than highlighting the terminal). So am not sure if there's any configurations that would be required to enable mouse click events in the first place or if it is something missing in the webLinksAddon 
constructor/code initialization. 

Version used:
```
    "xterm": "^5.1.0",
    "xterm-addon-web-links": "^0.8.0",
```
Initializing Terminal:
```
var terminal = new Terminal({
    linkHandler: {
      // If you explicitly support and sanitize the links
      allowNonHttpProtocols: true
    },
    screenKeys: true,
    mouseEvents: true,
    useStyle: true,
    cursorBlink: true,
    fullscreenWin: true,
    maximizeWin: true,
    screenReaderMode: true,
    allowProposedApi: true,
    rows: numRows,
    cols: numCols, 
  });
```

Weblink AddOn constructor: 
```
  var webLinksAddon = new WebLinksAddon.WebLinksAddon(    
      (event, url) => {
        console.log(url);
        //window.open(url);
        openExternal(uri);
        if (event.altKey) shell.openExternal(uri);
      },
      {
        tooltipCallback: () => console.log("hello")
      }
  );
  terminal.loadAddon(webLinksAddon);
```

Sample code tested:
```
  terminal.writeln('\x1b[3mLinks\x1b[0m');
  // simple link
  terminal.writeln('click here: https://xtermjs.org/');
  // OSC 8 hyperlink escape codes
  terminal.writeln('click here: \x1b]8;;https://xtermjs.org\x07OSC 8 hyperlink text\x1b]8;;\x07');
  terminal.writeln('\e]8;;http://www.google.com\aThis is a link\e]8;;\a');
```
Here's the output we see (the grey area is the result of a mouse click).
![image](https://user-images.githubusercontent.com/4099578/222000259-6030a4dc-d3d2-4d14-add3-9e26fd0b9913.png)

Note: not looking at the registerLinkProvider hardcoded option, though this too did not enable the links.

Thanks in advance for any help/guidance in the right direction..

**Comment by Tyriar** [comment]
Are you using an old version of the web links addon? The latest doesn't have a tooltip callback:

https://github.com/xtermjs/xterm.js/blob/5b05d4500b30ec5cfa3140d8674f4059741c40f8/addons/xterm-addon-web-links/typings/xterm-addon-web-links.d.ts#L9-L54

**Reply by isubasin** [reply]
Hi @Tyriar ,  Thanks for checking on this for us.

The application is developed using GoLang and it has the dependency upgraded to use "xterm-addon-web-links": "^0.8.0" and "xterm": "^5.1.0". So the app was build with this version from https://registry.npmjs.org/xterm-addon-web-links/-/xterm-addon-web-links-0.8.0.tgz. 

The code snippet for the tooltip was added during one of the tests, oddly this did not throw any exceptions.

Looking at https://github.com/xtermjs/xterm.js/blob/5b05d4500b30ec5cfa3140d8674f4059741c40f8/addons/xterm-addon-web-links/src/WebLinksAddon.ts however, seems we have the same version code on app run time source for WebLinksAddon.

**Comment by RahjIII** [comment]
I just opened a similar discussion, I'm sorry I didn't see this one first.

Try setting screenReaderMode to false and see if that enables the links?  If so... same issue I'm having.

**Reply by isubasin** [reply]
Hey @RahjIII .. Thank you, we didn't realize this was caused by the screenReaderMode - thank you. We are able to get the links to work with the screenReaderMode disabled, however, like you , we too need the accessibility tree. 

--> one more issue even with screenReaderMode off, : For ASCII character UI links are not clickable - we can only highlight , right click and select the GO TO <links> option. Is that the current expected behavior ?
![image](https://user-images.githubusercontent.com/4099578/223172424-9048fdda-483e-44a9-a84a-b289441689a9.png)

**Reply by jerch** [reply]
Mabe try with lowercase? While uppercase is only discouraged (thus not forbidden) in some parts of an URI, the canonical form is always lowercase. Uppercase creates alot of parsing fuzz, which the addon does not support currently.

**Reply by isubasin** [reply]
Thank you @jerch -- Yes this did make the links clickable.. unfortunately this CHUI system takes in most of user inputs in UPPERCASE except for emails. This is good progress for us so far - thank you.

**Comment by Tyriar** [comment]
This problem was fixed in https://github.com/xtermjs/xterm.js/pull/4491
